src/macros.rs

```diff
@@ -80,3 +80,33 @@ macro_rules! type_row {
 
 #[allow(unused_imports)]
 pub use type_row;
+
+/// Declare 'const' variables holding new ExtensionIds, validating
+/// that they are well-formed as separate tests - hence, usable at the top level
+/// of a test module only. Example:
+/// ```rust
+/// # mod test {
+/// # use hugr::macros::test_const_ext_id;
+///   test_const_ext_id! {
+///     pub const EXT_A: ExtensionId = "A";
+///     #[cfg(foobar)] pub (super) const EXT_A_B: ExtensionId = "A.B";
+///     const EXT_BAD: ExtensionId = "..55"; // this will generate a failing #[test] fn ....
+///   }
+/// # }
+/// ```
+#[macro_export]
+macro_rules! test_const_ext_id {
```

Comment:
Since the main effect of the macro is declaring a cost, maybe rename it to `const_extension_id` ?

The docs already explain that it does the validation.

Reply:
Yes, fair - it actually works in prod code too, if you fix the `paste` dep (which we won't). I went with `const_extension_ids` as you can declare more than one...